deepchecks/tabular/context.py

```diff
@@ -63,32 +63,41 @@ class _DummyModel:
     predictions: pd.DataFrame
     proba: pd.DataFrame
 
-    def __init__(self,
-                 test: Dataset,
-                 y_proba_test: t.Optional[np.ndarray] = None,
-                 y_pred_test: t.Optional[np.ndarray] = None,
-                 train: t.Union[Dataset, None] = None,
-                 y_pred_train: t.Optional[np.ndarray] = None,
-                 y_proba_train: t.Optional[np.ndarray] = None,
-                 validate_data_on_predict: bool = True,
-                 model_classes: t.Optional[t.List] = None):
-
+    def __init__(
+        self,
+        test: Dataset,
+        y_proba_test: t.Optional[np.ndarray] = None,
+        y_pred_test: t.Optional[np.ndarray] = None,
+        train: t.Optional[Dataset] = None,
+        y_pred_train: t.Optional[np.ndarray] = None,
+        y_proba_train: t.Optional[np.ndarray] = None,
+        validate_data_on_predict: bool = True,
+        model_classes: t.Optional[t.List[t.Any]] = None
+    ):
+        # TODO:
+        # constructor signature says that test cannot be `None`
+        # why do we check it then?
         if train is not None and test is not None:
             # check if datasets have same indexes
-            if set(train.data.index) & set(test.data.index):
-                train.data.index = map(lambda x: f'train-{x}', list(train.data.index))
-                test.data.index = map(lambda x: f'test-{x}', list(test.data.index))
-                get_logger().warning('train and test datasets have common index - adding "train"/"test"'
-                                     ' prefixes. To avoid that provide datasets with no common indexes '
-                                     'or pass the model object instead of the predictions.')
+            train_index = train.data.index
```

Comment:
redundant, no need to create those variables - they have no further use

Reply:
used twice so I'll keep them